src/main/webapp/app/tasks/task.service.ts

```diff
@@ -28,5 +42,5 @@ export interface TaskService {
    * @param id the ID of the task to be removed
    * @returns an empty `Observable`
    */
-  delete(id: string): Observable<void>;
+  delete(id: string, statusIndex?: number): Observable<void>;
```

Comment:
Status index shouldn't be optional as you will get an error when calling the actual function in the local task service (an other way would be to implement the service function handling that)

Reply:
I got an error from default-task-service.ts I didn't want to change or delete it so I preferred to do optional. I didn't think another way. I could solve it as you say.